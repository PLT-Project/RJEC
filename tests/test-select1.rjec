func foo(a chan int, b chan int, c chan int) {
    for {
        select {
        case i := <- a:
            printi(i);
        case j := <- b:
            printi(j);
        case k := <- c:
            printi(k);
        }
    }
}

func main() {
    a := make(chan int);
    b := make(chan int);
    c := make(chan int);
    yeet foo(a, b, c);
    a <- 777;
    b <- 888;
    c <- 1313;
}