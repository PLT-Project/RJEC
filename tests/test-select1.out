select: got index 0
-1
select: got index 1
-1